heron/proto/topology.proto

```diff
@@ -69,7 +75,10 @@ message Config {
     required string key = 1;
     // Only one of the below are set
     optional string value = 2;
-    optional bytes java_serialized_value = 3;
+    //optional bytes java_serialized_value = 3;
+    optional bytes serialized_value = 3;
+
+    required ConfigValueType type = 4;
```

Comment:
@billonahill I think these objects are created when a topology is submitted and other services (tracker/ui) reference them, so it's safe to be backward compatible.  (Also, for a similar reason, required/optioanl fields were [removed in protobuf 3.0](http://stackoverflow.com/questions/31801257/why-required-and-optional-is-removed-in-protocol-buffers-3)) 
I just pushed a version where ConfigValueType is optional --- It shouldn't affect the existing codes, and when we add a multi-lang support later, we can possibly make it backward compatible by 1) check the existence of ConfigValueType, and if it exists, handle according to that type, 2) if not existent, handle as if the value is Java serialized.